src/Type/Schema.php

```diff
@@ -524,22 +524,24 @@ public function isPossibleType(AbstractType $abstractType, ObjectType $possibleT
      */
     public function isSubType(AbstractType $abstractType, ImplementingType $maybeSubType): bool
     {
+        if ($abstractType instanceof InterfaceType) {
+            return $maybeSubType->implementsInterface($abstractType);
+        }
+
+        if ($abstractType instanceof UnionType) {
+            return $abstractType->isPossibleType($maybeSubType);
+        }
+
```

Comment:
I think we do not need the other code, and can throw here with a message reminding future developers of this library to handle a possible 3rd `AbstractType` (not needed now).

Reply:
Wouldn't you not throw so that you can still send impossible types to this method and it just returning false for types it cannot handle? Since technically users could also use this method (it's part of `@api`).

Reply:
`AbstractType` must be a `UnionType` or `InterfaceType`. It is not part of `@api` so technically users should not implement it 😉 In more capable type systems (e.g. TypeScript), `AbstractType` is modeled as a Union type and thus impossible to use wrong.

I think returning `false` here is failing silently.

Reply:
Yep, okay I agree, something something PHP 8 right 😉 I'll change it to throw!

Reply:
Yeah, PHP 8 would allow defining the argument type as `UnionType | InterfaceType`. Still not quite as ergonomic as TypeScript - perhaps https://wiki.php.net/rfc/union_types_v2#type_aliases will be picked up at some point 🤞